bundles/framework/publisher2/view/PanelToolLayout.js

```diff
@@ -159,7 +159,7 @@ Oskari.clazz.define('Oskari.mapframework.bundle.publisher2.view.PanelToolLayout'
             var me = this;
             this.tools.forEach(tool => {
                 // don't call for tools that already have been set enabled (=plugin has already been created.)
-                if (tool.isDisplayed(me.data) && !tool.isShownInToolsPanel() && !tool.state.enabled) {
+                if (tool.isDisplayed(me.data) && !tool.isShownInToolsPanel() && (tool.state && !tool.state.enabled)) {
                     tool.setEnabled(true);
                 }
```

Comment:
Maybe we should try if we can just remove this loop completely. It feels like it's used to fix something that is out of sync by calling setEnabled() for tools that are already "enabled". Let's fix the broken tools instead and get rid of this loop.